src/bun.js/bindings/bindings.cpp

```diff
@@ -153,7 +153,7 @@ static bool canPerformFastPropertyEnumerationForIterationBun(Structure* s)
 }
 
 // adapted from underscorejs [https://underscorejs.org/docs/modules/isEqual.html]
-bool Bun__deepEquals(JSC__JSGlobalObject* globalObject, JSValue v1, JSValue v2, Vector<std::pair<JSC::JSValue, JSC::JSValue>, 16>& stack, ThrowScope* scope, bool addToStack)
+bool Bun__deepEquals(JSC__JSGlobalObject* globalObject, JSValue v1, JSValue v2, Vector<std::pair<JSC::JSValue, JSC::JSValue>, 16>& stack, ThrowScope* scope, bool addToStack, bool strict)
```

Comment:
let's make this a 

`template<bool isStrict>`

and add `constrexpr` around any place that needs to only run for strict check

or just have a separate `Bun__strictEquals`